mathics/builtin/files.py

```diff
@@ -4942,3 +4958,211 @@ def apply(self, context, evaluation):
             return SymbolFailed
 
         return SymbolNull
+
+
+class URLSave(Builtin):
+    """
+    <dl>
+    <dt>'URLSave["url"]'
+        <dd>Save "url" in a temporary file.
+    <dt>'URLSave["url", $filename$]'
+        <dd>Save "url" in $filename$.
+    </dl>
+    """
+    messages = {"invfile": '`1` is not a valid Filename',
+                "invhttp": '`1` is not a valid URL'
+                }
+    def apply_1(self, url, evaluation, **options):
+        'URLSave[url_String, OptionsPattern[URLSave]]'
+        return self.apply_2(url, None, evaluation, **options)
+
+    def apply_2(self, url, filename, evaluation, **options):
```

Comment:
Something to think about for the future. I know the ocnvention we've been using is apply with a number afterwards, but I think things would be more meaningful if put something more descriptive in there like `apply_with_filename_parameter()`